nav2_costmap_2d/plugins/voxel_layer.cpp

```diff
@@ -400,7 +401,8 @@ void VoxelLayer::raytraceFreespace(
     clearing_endpoints_.header.frame_id = global_frame_;
     clearing_endpoints_.header.stamp = clearing_observation.cloud_->header.stamp;
 
-    clearing_endpoints_pub_->publish(clearing_endpoints_);
+    auto cloud_msg = std::make_unique<sensor_msgs::msg::PointCloud>(std::move(clearing_endpoints_));
```

Comment:
Why is this move required?

Reply:
Actually, `clearing_endpoints_` is not being used anywhere outside the function `VoxelLayer::raytraceFreespace`. I can try converting it to a unique_ptr local to this function.